### ISSUE TITLE
Missing diagnostics "warning: cv qualifiers ignored for a base class"

### ISSUE THREAD

**Issue by fsb4000** [llvm/llvm-project#55474]
http://eel.is/c++draft/class.derived.general#2.sentence-2

Everything works correctly but such diagnostic would be good.

```c++
class Base { 
public:
   int base;
};

typedef const Base ConstBase;

class CDerived: public ConstBase {
public:
   int derived;
};
```

https://godbolt.org/z/1ns8Tc7Ko

**Comment by manoj-al** [comment]
Hi, 
I wanted to fix this issue, is -Wignored-qualifiers is the right flag to add this warning?
Thanks.

**Comment by fsb4000** [comment]
> is -Wignored-qualifiers is the right flag to add this warning?

Yes, I think it would be great!

**Comment by a-tarasyuk** [comment]
@AaronBallman @erichkeane should all of the following qualifiers be handled?

https://github.com/llvm/llvm-project/blob/1e492285f33ca7c7efa97671990d47a32eaf31f7/clang/lib/Sema/SemaType.cpp#L2885-L2889

**Comment by erichkeane** [comment]
> @AaronBallman @erichkeane should all of the following qualifiers be handled?
> 
> https://github.com/llvm/llvm-project/blob/1e492285f33ca7c7efa97671990d47a32eaf31f7/clang/lib/Sema/SemaType.cpp#L2885-L2889

I would think ANY qualifier would be ignored in this case and should be diagnosed. That said, I'm not sure which of those would actually be able to be used, so testing them might be difficult.
